src/Curl/Curl.php

```diff
@@ -105,9 +105,11 @@ public function __construct($base_url = null)
         $this->setDefaultUserAgent();
         $this->setDefaultJsonDecoder();
         $this->setDefaultTimeout();
-        $this->setOpt(CURLINFO_HEADER_OUT, true);
-        $this->setOpt(CURLOPT_HEADERFUNCTION, array($this, 'headerCallback'));
-        $this->setOpt(CURLOPT_RETURNTRANSFER, true);
+        $this->setOpts($options + array(
+            CURLINFO_HEADER_OUT => true,
+            CURLOPT_HEADERFUNCTION => array($this, 'headerCallback'),
+            CURLOPT_RETURNTRANSFER => true
+        ));
```

Comment:
I believe we should set default options first. Then apply the user's preferred options after.

```php
        $this->setOpt(CURLINFO_HEADER_OUT, true);
        $this->setOpt(CURLOPT_HEADERFUNCTION, array($this, 'headerCallback'));
        $this->setOpt(CURLOPT_RETURNTRANSFER, true);
        $this->setOpts($options);
```